src/Microsoft.ML.OnnxTransform/OnnxTransform.cs

```diff
@@ -0,0 +1,445 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.CommandLine;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.EntryPoints;
+using Microsoft.ML.Runtime.Internal.Utilities;
+using Microsoft.ML.Runtime.Model;
+using Microsoft.ML.Scoring;
+using Microsoft.ML.OnnxScoring;
+using OnnxShape = System.Collections.Generic.List<long>;
+using Microsoft.ML.Core.Data;
+using Microsoft.ML.Data.StaticPipe.Runtime;
+
+[assembly: LoadableClass(OnnxTransform.Summary, typeof(IDataTransform), typeof(OnnxTransform),
+    typeof(OnnxTransform.Arguments), typeof(SignatureDataTransform), OnnxTransform.UserName, OnnxTransform.ShortName)]
+
+[assembly: LoadableClass(OnnxTransform.Summary, typeof(IDataTransform), typeof(OnnxTransform),
+    null, typeof(SignatureLoadDataTransform), OnnxTransform.UserName, OnnxTransform.LoaderSignature)]
+
+[assembly: LoadableClass(typeof(OnnxTransform), null, typeof(SignatureLoadModel),
+    OnnxTransform.UserName, OnnxTransform.LoaderSignature)]
+
+[assembly: LoadableClass(typeof(IRowMapper), typeof(OnnxTransform), null, typeof(SignatureLoadRowMapper),
+    OnnxTransform.UserName, OnnxTransform.LoaderSignature)]
+
+[assembly: EntryPointModule(typeof(OnnxTransform))]
+
+namespace Microsoft.ML.OnnxScoring
+{
+    public sealed class OnnxTransform : ITransformer, ICanSaveModel
+    {
+        public sealed class Arguments : TransformInputBase
+        {
+
+            [Argument(ArgumentType.Required, HelpText = "Path to the onnx model file.", ShortName = "model", SortOrder = 0)]
+            public string ModelFile;
+
+            [Argument(ArgumentType.Multiple | ArgumentType.Required, HelpText = "TBD", SortOrder = 1)]
+            public string InputColumn;
+
+            [Argument(ArgumentType.Multiple | ArgumentType.Required, HelpText = "TBD", SortOrder = 2)]
+            public string OutputColumn;
+        }
+
+        private readonly IHost _host;
+        private readonly Arguments _args;
+        private readonly OnnxModel _model;
+        private const string RegistrationName = "OnnxTransform";
+
+        internal const string Summary = "Transforms the data using the Onnx model.";
+        internal const string UserName = "OnnxTransform";
+        internal const string ShortName = "Onnx";
+        internal const string LoaderSignature = "OnnxTransform";
+
+        public readonly string[] Inputs;
+        public readonly string[] Outputs;
```

Comment:
Do you plan to support multiple input /outputs or only one? 
It looks weird, you constantly cast single objects to arrays and back.

Reply:
It's single input/output transform. I cleaned all the casts.

---
In reply to: [219612560](https://github.com/dotnet/machinelearning/pull/942#discussion_r219612560) [](ancestors = 219612560)